tests/providers/amazon/aws/sensors/test_dynamodb.py

```diff
@@ -0,0 +1,129 @@
+                {"AttributeName": "SK", "KeyType": "RANGE"},
+            ],
+            AttributeDefinitions=[
+                {"AttributeName": "PK", "AttributeType": "S"},
+                {"AttributeName": "SK", "AttributeType": "S"},
+            ],
+            BillingMode="PAY_PER_REQUEST",
+        )
+        return dynamodb
+
+    return dynamodb_client
+
+
+class TestDynamoDBValueSensor:
+    def setup_method(self):
+        self.mock_context = MagicMock()
+
+    def test_init(self):
+        sensor = DynamoDBValueSensor(
+            task_id=TASK_ID,
+            table_name=TABLE_NAME,
+            partition_key_name="PK",
+            partition_key_value="Test",
+            sort_key_name="SK",
+            sort_key_value="2022-07-12T11:11:25-0400",
+            attribute_name="Foo",
+            attribute_value="Bar",
+            aws_conn_id=AWS_CONN_ID,
+            region_name=REGION_NAME,
+        )
+
+        assert TASK_ID == sensor.task_id
+        assert AWS_CONN_ID == sensor.aws_conn_id
+        assert REGION_NAME == sensor.region_name
+
+    @mock_dynamodb
+    def test_get_conn_returns_a_boto3_connection(self):
+        hook = DynamoDBHook(aws_conn_id=AWS_CONN_ID)
+        assert hook.get_conn() is not None
+
+    @mock_dynamodb
+    @mock.patch("airflow.providers.amazon.aws.sensors.dynamodb.DynamoDBHook")
+    def test_sensor_with_pk_and_sk(self, ddb_mock):
+        pprint(ddb_mock)
+
+        hook = DynamoDBHook(
+            aws_conn_id=AWS_CONN_ID, table_name=TABLE_NAME, table_keys=["PK"], region_name=REGION_NAME
+        )
+
+        hook.get_conn().create_table(
+            TableName=TABLE_NAME,
+            KeySchema=[
+                {"AttributeName": "PK", "KeyType": "HASH"},
+            ],
+            AttributeDefinitions=[{"AttributeName": "PK", "AttributeType": "S"}],
+            ProvisionedThroughput={"ReadCapacityUnits": 10, "WriteCapacityUnits": 10},
+        )
+
+        table = hook.get_conn().Table(TABLE_NAME)
+        table.meta.client.get_waiter("table_exists").wait(TableName=TABLE_NAME)
+
+        assert table.table_status == "ACTIVE"
+
+        sensor = DynamoDBValueSensor(
+            task_id=TASK_ID,
+            poke_interval=30,
+            timeout=120,
+            soft_fail=False,
+            retries=10,
+            table_name=TABLE_NAME,  # replace with your table name
+            partition_key_name="PK",  # replace with your partition key name
+            partition_key_value="Test",  # replace with your partition key value
+            sort_key_name="SK",  # replace with your sort key name (if applicable)
+            sort_key_value="2023-03-28T11:11:25-0400",  # replace with your sort key value (if applicable)
+            attribute_name="Foo",  # replace with the attribute name to wait for
+            attribute_value="Bar",  # replace with the attribute value to wait for (sensor will return true when this value matches the attribute value in the item)
+        )
+
```

Comment:
For troubleshooting your issue:  if you add `assert not poke(None)` here before the `table.put_item` it should pass that assert, right?

Reply:
Correct, `assert not sensor.poke(None)` allows the test to pass.

Reply:
You can resolve this thread so it's easier to see the new test in one chunk :+1: